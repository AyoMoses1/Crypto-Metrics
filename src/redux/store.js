import { configureStore } from '@reduxjs/toolkit';
import cryptoReducer from './cryptos/cryptoSlice';

const store = configureStore({
  reducer: {
    cryptoReducer,
  },
});

export default store;
