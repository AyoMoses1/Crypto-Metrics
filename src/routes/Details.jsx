/*eslint-disable*/
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useLocation } from 'react-router-dom';
import Card from 'react-bootstrap/Card';
import { Container } from 'react-bootstrap';
import ListGroup from 'react-bootstrap/ListGroup';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { fetchCountryDetails } from '../redux/cryptos/cryptoSlice';

function Details() {
  const dispatch = useDispatch();
  const { state } = useLocation();
  const res = useSelector((state) => state.cryptoReducer.details);

  useEffect(() => {
    dispatch(fetchCountryDetails(state));
  }, [dispatch, state]);

  return (
    <Container fluid className="margin-top">
      <div className='banner' style={{background: `url(${res?.countryInfo?.flag}`}}>
        <h2>Covid Tracker</h2>
        <p>Over 200views</p>
      </div>
      <Row className="justify-content-center">
        <Col md={{ span: 4 }} sm={{ span: 12 }} className="mb-4">
          <div className='grid-list'>
            <h4>Cases</h4>
            <p>{res?.cases}</p>
          </div>
          <div className='grid-list'>
            <h4>Deaths</h4>
            <p>{res?.deaths}</p>
          </div>
          <div className='grid-list'>
            <h4>Recovered</h4>
            <p>{res?.recovered}</p>
          </div>
          <div className='grid-list'>
            <h4>Population</h4>
            <p>{res?.population}</p>
          </div>
        </Col>
      </Row>
    </Container>
  );
}

export default Details;

