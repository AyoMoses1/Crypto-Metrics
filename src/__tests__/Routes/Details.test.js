import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import Details from '../../routes/Details';

it('App component test', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <Details />
        </Provider>
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
