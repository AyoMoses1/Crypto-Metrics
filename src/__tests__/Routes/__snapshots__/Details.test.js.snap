// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test 1`] = `
<div
  className="margin-top container-fluid"
>
  <a
    className="nav-link"
    disabled={false}
    href="/"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      <i
        className="fa-solid fa-arrow-left"
      />
    </span>
  </a>
  <div
    className="banner"
    style={
      Object {
        "background": "url(undefined",
      }
    }
  >
    <h2>
      Covid Tracker
    </h2>
    <p>
      Over 200views
    </p>
  </div>
  <div
    className="justify-content-center row"
  >
    <div
      className="mb-4 col-md-4 col-sm-12"
    >
      <div
        className="grid-list"
      >
        <h4>
          Cases
        </h4>
        <p />
      </div>
      <div
        className="grid-list"
      >
        <h4>
          Deaths
        </h4>
        <p />
      </div>
      <div
        className="grid-list"
      >
        <h4>
          Recovered
        </h4>
        <p />
      </div>
      <div
        className="grid-list"
      >
        <h4>
          Population
        </h4>
        <p />
      </div>
    </div>
  </div>
</div>
`;
